### ISSUE TITLE
`DBQueryExecute`

### ISSUE THREAD

**Issue by bmyerz** [uwescience/raco#507]
@mbalazin 
Implement the ability to pass arbitrary commands to Postgres. Add:
- a executePostgres sort of syntax to MyriaL
- an operator `DBQueryExecute` that is like `DBQueryScan` except a sink.

**Comment by bmyerz** [comment]
@mbalazin @senderista  Post-meeting, it sounded like the consensus was to expose the `DBQueryExecute` operator up to MyriaL only for create index and create materialized view.

**Comment by senderista** [comment]
The preliminary implementation will support only `CREATE VIEW`, `CREATE MATERIALIZED VIEW`, `CREATE INDEX` in MyriaL and will not accept arbitrary Postgres DDL. (`CREATE TABLE AS` should probably be used to implement `CREATE MATERIALIZED VIEW` since we don't need auto-updating views.) We need to add corresponding objects to the Myria catalog and mark them as in-progress during creation, so the operations will be idempotent (objects will be created only once on each partition despite repeated invocations), and in-progress objects will not be visible to the catalog APIs.

More generally, we could support arbitrary DDL, but since we can't track arbitrary Postgres entities in the Myria catalog, we need to execute the DDL in a distributed transaction to guarantee atomicity (e.g., http://www.postgresql.org/docs/9.4/static/sql-prepare-transaction.html). This will not be trivial.

**Comment by bmyerz** [comment]
Will the Raco/MyriaX operator be create view / create materialized view or DBQueryExecute?

**Comment by senderista** [comment]
Possibly we can just translate the create view/index statements into a generic `DbQueryExecute` operator, although that doesn't address how the coordinator enforces consistency guarantees (which would have to be done either in a specialized operator or in a REST API implementation). I would like some feedback from @jingjingwang on this question.

**Comment by senderista** [comment]
Created https://github.com/uwescience/myria/issues/816 for the `DbQueryExecute` operator.

**Comment by senderista** [comment]
I think we decided we're not interested in exposing arbitrary DDL in MyriaL. https://github.com/uwescience/myria/issues/816 has already been resolved (we decided to expose CreateView/CreateIndex/CreateFunction in REST APIs rather than MyriaL for now).

**Comment by senderista** [comment]
Also this is now superseded by https://github.com/uwescience/raco/issues/538
